src/dsp_tools/xmllib/helpers.py

```diff
@@ -56,3 +64,292 @@ def _name_label_mapper_iterator(
+            return None
+        return f"GREGORIAN:BC:{start_year}:BC:{end_year}"
+
+    single_year_regex = rf"{lookbehind}({year_regex}) {french_bc_regex}{lookahead}"
+    single_year = regex.search(single_year_regex, string)
+    if single_year:
+        start_year = int(single_year.group(1))
+        return f"GREGORIAN:BC:{start_year}:BC:{start_year}"
+
+    return None
+
+
+def _from_iso_date(iso_date: Match[str]) -> str:
+    year = int(iso_date.group(1))
+    month = int(iso_date.group(2))
+    day = int(iso_date.group(3))
+    date = datetime.date(year, month, day)
+    return f"GREGORIAN:CE:{date.isoformat()}:CE:{date.isoformat()}"
+
+
+def _expand_2_digit_year(year: int) -> int:
+    current_year = datetime.date.today().year - 2000
+    if year <= current_year:
+        return year + 2000
+    elif year <= 99:
+        return year + 1900
+    else:
+        return year
+
+
+def _from_eur_date_range(eur_date_range: Match[str]) -> str:
+    startday = int(eur_date_range.group(1))
+    startmonth = int(eur_date_range.group(2)) if eur_date_range.group(2) else int(eur_date_range.group(5))
+    startyear = int(eur_date_range.group(3)) if eur_date_range.group(3) else int(eur_date_range.group(6))
+    startyear = _expand_2_digit_year(startyear)
+    endday = int(eur_date_range.group(4))
+    endmonth = int(eur_date_range.group(5))
+    endyear = int(eur_date_range.group(6))
+    endyear = _expand_2_digit_year(endyear)
+    startdate = datetime.date(startyear, startmonth, startday)
+    enddate = datetime.date(endyear, endmonth, endday)
+    if enddate < startdate:
+        raise ValueError
+    return f"GREGORIAN:CE:{startdate.isoformat()}:CE:{enddate.isoformat()}"
+
+
+def _from_eur_date(eur_date: Match[str]) -> str:
+    startday = int(eur_date.group(1))
+    startmonth = int(eur_date.group(2))
+    startyear = int(eur_date.group(3))
+    startyear = _expand_2_digit_year(startyear)
+    date = datetime.date(startyear, startmonth, startday)
+    return f"GREGORIAN:CE:{date.isoformat()}:CE:{date.isoformat()}"
+
+
+def _from_monthname_date(monthname_date: Match[str]) -> str:
+    day = int(monthname_date.group(2))
+    month = _months_dict[monthname_date.group(1)]
+    year = int(monthname_date.group(3))
+    date = datetime.date(year, month, day)
+    return f"GREGORIAN:CE:{date.isoformat()}:CE:{date.isoformat()}"
+
+
+def _from_year_range(year_range: Match[str]) -> str:
+    startyear = int(year_range.group(1))
+    endyear = int(year_range.group(2))
+    if endyear // 10 == 0:
+        # endyear is only 1-digit: add the first 2-3 digits of startyear
+        endyear = startyear // 10 * 10 + endyear
+    elif endyear // 100 == 0:
+        # endyear is only 2-digit: add the first 1-2 digits of startyear
+        endyear = startyear // 100 * 100 + endyear
+    if endyear <= startyear:
+        raise ValueError
+    return f"GREGORIAN:CE:{startyear}:CE:{endyear}"
+
+
+def make_xsd_compatible_id(input_value: str | float | int | bool) -> str:
```

Comment:
I think it does not hurt to also accept other data types as long as they are not empty

Reply:
While I can see a potential use case for integers and floats, the boolean probably doesn't make sense.